sympy/polys/polymatrix.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import print_function
+
+from sympy.core import S
+from sympy.matrices.dense import DenseMatrix
+from sympy.matrices.matrices import classof
+
+
+class PolyMatrix(DenseMatrix):
+    """
+    Matrix of polynomials.
+
+    >>> from sympy.polys.polymatrix import PolyMatrix
+    >>> from sympy import Symbol, Matrix, Poly
+    >>> x = Symbol('x')
+    >>> pm1 = PolyMatrix([[Poly(x**2, x), Poly(-x, x)], [Poly(x**3, x), Poly(-1 + x, x)]])
+    >>> v1 = Matrix([[1, 0], [-1, 0]])
+    >>> pm1*v1
+    Matrix([
+    [    Poly(x**2 + x, x, domain='ZZ'), Poly(0, x, domain='ZZ')],
+    [Poly(x**3 - x + 1, x, domain='ZZ'), Poly(0, x, domain='ZZ')]])
+
+    >>> v1*pm1
+    Matrix([
+    [ Poly(x**2, x, domain='ZZ'),  Poly(-x, x, domain='ZZ')],
+    [-Poly(x**2, x, domain='ZZ'), -Poly(-x, x, domain='ZZ')]])
+
+    >>> pm2 = PolyMatrix([[Poly(x**2, x, domain='QQ'), Poly(0, x, domain='QQ'), Poly(-x**2, x, domain='QQ'), \
+            Poly(x**3, x, domain='QQ'), Poly(0, x, domain='QQ'), Poly(-x**3, x, domain='QQ')]])
+    >>> v2 = Matrix([1, 0, 0, 0, 0, 0])
+    >>> pm2*v2
+    Matrix([[Poly(x**2, x, domain='QQ')]])
+
+    """
+    _class_priority = 10
+    _sympify = staticmethod(lambda x: x)
+
+    def __new__(cls, *args, **kwargs):
```

Comment:
Yes, which it is right now. I verbatim copied the code from `MutableDenseMatrix`.